pygeo/geo_utils/ffd_generation.py

```diff
@@ -0,0 +1,293 @@
+            for k in range(size[1]):
+                U = P + np.outer(d1, (R - P))[k]
+                V = Q + np.outer(d1, (S - Q))[k]
+                X[j, k] = U + np.outer(d2, (V - U))
+
+        for dim in range(3):
+            line = ""
+            for k in range(Nk):
+                for j in range(Nj):
+                    for i in range(Ni):
+                        idc = [-1, -1, -1]
+                        idc[axes.index("i")] = i
+                        idc[axes.index("j")] = j
+                        idc[axes.index("k")] = k
+                        line += f"{X[idc[0], idc[1], idc[2], dim]: .4e}\t"
+                        if len(line) + 11 > 80:
+                            f.write(line + "\n")
+                            line = ""
+            if len(line) > 0:
+                f.write(line + "\n")
+
+    f.close()
+
+
+def createFittedWingFFD(surf, surfFormat, outFile, leList, teList, nSpan, nChord, absMargins, relMargins, liftIndex):
+
+    """
+    Generates a wing FFD with chordwise points that follow the airfoil geometry.
+
+    Parameters
+    ----------
+    surf : pyGeo object or list or str
+        The surface around which the FFD will be created.
+        See the documentation for :meth:`pygeo.DVConstraints.setSurface` for details.
+
+    surfFormat : str
+        The surface format.
+        See the documentation for :meth:`pygeo.DVConstraints.setSurface` for details.
+
+    outFile : str
+        Name of output file written in PLOT3D format.
+
+    leList : list or array
+        List or array of points (of size Nx3 where N is at least 2) defining the 'leading edge'.
+
+    teList : list or array
+        Same as leList but for the trailing edge.
+
+    nSpan : int or list of int
+        Number of spanwise sections in the FFD.
+        Use a list of length N-1 to specify the number for each segment defined by leList and teList
+        and to precisely match intermediate locations.
+
+    nChord : int
+        Number of chordwise points in the FFD.
+
+    absMargins : list of float
+        List with 3 items specifying the absolute margins in the [chordwise, spanwise, thickness] directions.
+
+    relMargins : list of float
+        List with 3 items specifying the relative margins in the [chordwise, spanwise, thickness] directions.
+        Relative margins are applied as a fraction of local chord, span, and local thickness.
+
+    liftIndex : int
+        Index specifying which direction lift is in (same as the ADflow option).
+        Either 2 for the y-axis or 3 for the z-axis.
+        This is used to determine the wing's spanwise direction.
+
+    Examples
+    --------
+    >>> CFDSolver = ADFLOW(options=aeroOptions)
+    >>> surf = CFDSolver.getTriangulatedMeshSurface()
+    >>> surfFormat = "point-vector"
+    >>> outFile = "wing_ffd.xyz"
+    >>> nSpan = [4, 4]
+    >>> nChord = 8
+    >>> relMargins = [0.01, 0.001, 0.01]
+    >>> absMargins = [0.05, 0.001, 0.05]
```

Comment:
I am a bit confused about the usage of relative and absolute margins. Are we taking the most conservative option? A one-line explanation in the docstrings above would help.